WordPressKit/HTTPClient.swift

```diff
@@ -34,32 +35,57 @@ extension URLSession {
             return .failure(.unparsableResponse(response: nil, body: body))
         }
 
+        guard acceptableStatusCodes.contains(where: { $0 ~= response.statusCode }) else {
+            return .failure(.unacceptableStatusCode(response: response, body: body))
+        }
+
         return .success(.init(response: response, body: body))
     }
 
 }
 
-extension Result where Success == HTTPAPIResponse<Data> {
-
-    func assessStatusCode<S, E: LocalizedError>(
-        acceptable: [ClosedRange<Int>] = [200...299],
-        success: (Success) -> S?,
-        failure: (Success) -> E?
-    ) -> WordPressAPIResult<S, E> where Failure == WordPressAPIError<E> {
-        flatMap { response in
-            if acceptable.contains(where: { $0 ~= response.response.statusCode }) {
-                if let result = success(response) {
-                    return .success(result)
-                } else {
-                    return .failure(.unparsableResponse(response: response.response, body: response.body))
-                }
-            } else {
-                if let endpointError = failure(response) {
-                    return .failure(.endpointError(endpointError))
+extension WordPressAPIResult {
+
+    func mapSuccess<NewSuccess, E: LocalizedError>(
+        _ transform: (Success) -> NewSuccess?
+    ) -> WordPressAPIResult<NewSuccess, E> where Success == HTTPAPIResponse<Data>, Failure == WordPressAPIError<E> {
+        flatMap { success in
+            guard let newSuccess = transform(success) else {
+                return .failure(.unparsableResponse(response: success.response, body: success.body))
+            }
+
+            return .success(newSuccess)
+        }
+    }
+
+    func mapSuccess<NewSuccess: Decodable, E: LocalizedError>(
+        _ decoder: JSONDecoder = JSONDecoder()
```

Comment:
How do you feel about calling this `decodeSuccess`?

The motivation for this suggestion comes from looking at the usage in the unit tests:

```swift
        let result: WordPressAPIResult<Post, TestError> = await URLSession.shared
            .perform(request: .init(url: URL(string: "https://wordpress.org/hello")!))
            .mapSuccess()
```

Seeing that `mapSuccess` call with no argument was surprising as I'm used to seeing map-like calls taking passing a function or a closure.

Reply:
Renamed in https://github.com/wordpress-mobile/WordPressKit-iOS/pull/668/commits/cab530e224bdbcac065e48314d1b05688f01bbf3.